dev/build_release.py

```diff
@@ -211,28 +212,62 @@ def determine_gradle_root(self, name):
-        target [string]: The gradle build target.
+    This function runs a 'candidate' release using the last git tag as the package
+    version and the Bintray configuration passed through arguments. A 'candidate'
+    release builds the source, packages the debian and jar files, and publishes
+    those to the respective Bintray '$org/$repository'.
 
-      Returns:
-        BackgroundProcess
-      """
-      extra_args = []
-      if name == 'deck' and not 'CHROME_BIN' in os.environ:
-        extra_args.append('-PskipTests')
-
-      # Currently spinnaker is in a separate location
-      gradle_root = self.determine_gradle_root(name)
-      print 'Building {name}...'.format(name=name)
-      return BackgroundProcess.spawn(
-          'Building {name}'.format(name=name),
-          'cd "{gradle_root}"; ./gradlew {target} {extra}'.format(
-              gradle_root=gradle_root, target=target,
-              extra=' '.join(extra_args)))
+    The 'candidate' gradle release task throws a 409 if the package we are trying to
+    publish already exists. We'll publish unique package version using build
+    numbers. These will be transparent to end users since the only meaningful
+    version is the Spinnaker product version.
+
+    We will use -Prelease.useLastTag=true and ensure the last git tag is the
+    version we want to use. This tag has to be of the form 'X.Y.Z-$build' or
+    'vX.Y.Z-$build for gradle to use the tag as the version. This script will
+    assume that the source has been properly tagged to use the latest tag as the
+    package version for each component.
+
+    Args:
+      name [string]: Name of the subsystem repository.
+
+    Returns:
+      BackgroundProcess
+    """
+    jarRepo = self.__options.jar_repo
+    parts = self.__options.bintray_repo.split('/')
+    if len(parts) != 2:
+      raise ValueError(
+          'Expected --bintray_repo to be in the form <owner>/<repo>')
+    org, packageRepo = parts[0], parts[1]
+    bintray_key = os.environ['BINTRAY_KEY']
+    bintray_user = os.environ['BINTRAY_USER']
+
+    extra_args = [
+      '--stacktrace',
+      '-Prelease.useLastTag=true',
+      '-PbintrayPackageBuildNumber={number}'.format(number=self.__build_number),
+      '-PbintrayOrg="{org}"'.format(org=org),
+      '-PbintrayPackageRepo="{repo}"'.format(repo=packageRepo),
+      '-PbintrayJarRepo="{jarRepo}"'.format(jarRepo=jarRepo),
+      '-PbintrayKey="{key}"'.format(key=bintray_key),
+      '-PbintrayUser="{user}"'.format(user=bintray_user)
+    ]
+    if name == 'deck' and not 'CHROME_BIN' in os.environ:
+      extra_args.append('-PskipTests')
+
+    # Currently spinnaker is in a separate location
+    gradle_root = self.determine_gradle_root(name)
+    print 'Building and publishing {name}...'.format(name=name)
+    return BackgroundProcess.spawn(
+      'Building and publishing {name}...'.format(name=name),
+      'cd "{gradle_root}"; ./gradlew {extra} candidate'.format(
```

Comment:
`candidate` is the Nebula gradle task name. The only other choice that behaves the way we want is `final` at the moment, and that doesn't seem better. I'm not too sure about our ability to rename these.